pkg/service/roommanager.go

```diff
@@ -317,40 +317,50 @@ func (r *RoomManager) getOrCreateRoom(ctx context.Context, roomName livekit.Room
 		return nil, err
 	}
 
-	// construct ice servers
-	room = rtc.NewRoom(ri, *r.rtcConfig, &r.config.Audio, r.telemetry)
-	room.Hold()
+	r.lock.Lock()
+
+	currentRoom := r.rooms[roomName]
+	if currentRoom != lastSeenRoom {
+		r.lock.Unlock()
+		currentRoom.Hold()
```

Comment:
I'm not 100% confident about this line. Is it okay to assume it's `Hold`able? If this is not the case, can we just ignore it like this, hoping the outer logic would take care of it gracefully?

Reply:
It's highly unlikely that it'll fail here, but I think it's still better not assume that. Perhaps something like
```
if currentRoom.Hold() {
    return currentRoom, nil
} else {
    r.lock.Lock()
}
```

I don't love the lock handling here, but this would be the extreme case here.

Reply:
How about this one? https://github.com/livekit/livekit-server/pull/485/commits/8b8a264545d3f2733e64d33a29e42163d2c28b56

Reply:
so that would cause a deadlock in that path. Hold() also acquires the lock